src/app.ts

```diff
@@ -94,7 +95,9 @@ app.post('/execute', async (req: Request, res: Response) => {
   // const { firstCredentials, secondCredentials } = smRequest.config as any;
 
   // 2b. when using stateful skill, extract relevant credentials elsewhere (eg. memory) as config will not be present here
-  const { firstCredentials, secondCredentials } = smRequest.memory[0].value;
+  const [, credentials] = getMemoryValue(smRequest.memory, "credentials") as [boolean, any]
+  const firstCredentials = credentials.firstCredentials;
+  const secondCredentials = credentials.secondCredentials;
```

Comment:
this one can be simplified
```suggestion
  const { firstCredentials, secondCredentials } = credentials;
```